cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,545 @@ Cask Data Application Platform Release Notes
+
+- :cask-issue:`HYDRATOR-1252` - The SKipGram Feature Generator plugin is now available in
+  its own repository at `github.com/hydrator/skipgram-analytics 
+  <https://github.com/hydrator/skipgram-analytics>`__.
+
+- :cask-issue:`HYDRATOR-1253` - The Naive Bayes Analytics plugin is now available in its
+  own repository at `github.com/hydrator/naive-bayes-analytics
+  <https://github.com/hydrator/naive-bayes-analytics>`__.
+
+- :cask-issue:`HYDRATOR-1254` - The HashingTF Feature Generator plugin is now available in
+  its own repository at `github.com/hydrator/hashing-tf-feature-generator
+  <https://github.com/hydrator/hashing-tf-feature-generator>`__.
+
+- :cask-issue:`HYDRATOR-1255` - The LogisticRegression plugins are now available in their
+  own repository at `github.com/hydrator/logistic-regression-analytics
+  <https://github.com/hydrator/logistic-regression-analytics>`__.
+
+- :cask-issue:`HYDRATOR-1323` - Added a new ErrorTransform plugin-type that can be placed
+  after a pipeline stage to consume errors emitted by that stage.
+
+- :cask-issue:`HYDRATOR-1398` - Support added for Table datasets for lookups in plugins
+  and pipelines.
+
+
+Additional New Features
+.......................
+
+- :cask-issue:`CDAP-4556` - CDAP now uses environment variables in the ``spark-env.sh`` and
+  properties in the ``spark-defaults.conf`` when launching Spark programs.
+
+- :cask-issue:`CDAP-5107` - Added an HTTP RESTful endpoint to retrieve a specific property
+  for a specific version of an artifact in the ``system`` scope.
+
+- :cask-issue:`CDAP-8021` - Added the management of preferences at the application and
+  program levels.
+
+- :cask-issue:`CDAP-8122` - Made headers and the request/response bodies available in
+  audit logs for certain RESTful endpoints.
+
+- :cask-issue:`CDAP-8292` - Added support for CDH 5.10.0.
+
+Improvements
+------------
+
+- :cask-issue:`CDAP-3383` - Enabled in CDAP invalid transaction list pruning, a new
+  feature introduced in Apache Tephra. This automates the pruning of the invalid transaction
+  list after data for the invalid transaction has been dropped.
+
+- :cask-issue:`CDAP-6046` - Added an easier, additional syntax for the CDAP CLI
+  ``set/get/load/delete <type> preferences`` commands, with the preferences at the end of the
+  syntax, such as ``set workflow preferences MyApp.My.WF 'a=b c=d'``.
+
+- :cask-issue:`CDAP-7835` - The Metadata Service upgrades the metadata dataset to reduce
+  the time required by the upgrade tool during a CDAP upgrade.
+
+- :cask-issue:`CDAP-7962` - Program context information, including namespace, program
+  name, and program type, are now available in the MDC property of each ILoggingEvent
+  emitted from a program container.
+
+- :cask-issue:`CDAP-8019` - Added a configuration to control the timeout of CDAP Explore
+  operations.
+
+- :cask-issue:`CDAP-8061` - Moved the Cask Market Path to the ``cdap-defaults.xml`` file.
```

Comment:
"Users can now configure the path to their own private Cask Market using the configuration setting ``market.base.url``"